pkg/ipam/noop_allocator.go

```diff
@@ -0,0 +1,40 @@
+package ipam
+
+import (
+	"errors"
+	"net"
+)
+
+var errNotSupported = errors.New("Operation not supported")
+
+// noOpAllocator implements ipam.Allocator with no-op behavior.
+// It is used for IPAMDelegatedPlugin, where the CNI binary is responsible for assigning IPs
+// without relying on the cilium daemon or operator.
+type noOpAllocator struct{}
+
+func (n *noOpAllocator) Allocate(ip net.IP, owner string) (*AllocationResult, error) {
+	return nil, errNotSupported
+}
+
+func (n *noOpAllocator) AllocateWithoutSyncUpstream(ip net.IP, owner string) (*AllocationResult, error) {
+	return nil, errNotSupported
+}
+
+func (n *noOpAllocator) Release(ip net.IP) error {
+	return errNotSupported
+}
+
+func (n *noOpAllocator) AllocateNext(owner string) (*AllocationResult, error) {
+	return nil, errNotSupported
+}
+
+func (n *noOpAllocator) AllocateNextWithoutSyncUpstream(owner string) (*AllocationResult, error) {
+	return nil, errNotSupported
+}
+
+func (n *noOpAllocator) Dump() (map[string]string, string) {
+	return nil, ""
```

Comment:
tiny nit: The second `string` is used to display the IPAM status in `cilium status`. 

For example, here it's `24/34 allocated`.

```console
$ cilium status
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.21+ (v1.21.9-eks-0d102a7) [linux/amd64]
Cilium:                 Ok   1.10.10 (v1.10.10-024ecfc)
[...]
IPAM:                   IPv4: 24/34 allocated

```

We could return a static string like `"delegated to plugin"` here, instead of not showing anything.

Reply:
thanks! will update this